eth/CMakeLists.txt

```diff
@@ -2,15 +2,249 @@
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOprotocolConfigurator_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOproxy.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOproxy_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOreceiver.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOreceiver_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOropframe.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOropframe_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOrop.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOrop_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheBOARDtransceiver.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheBOARDtransceiver_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheFormer.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheFormer_hid.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheInfoDispatcher.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheInfoDispatcher_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheParser.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtheParser_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtransceiver.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtransceiver_hid.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtransmitter.h
+                                 ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/comm-v2/transport/EOtransmitter_hid.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/utils/eODeb_eoProtoParser.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/utils/eODeb_eoProtoParser_hid.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/utils/eOtheEthLowLevelParser.h
+#                                ${CMAKE_CURRENT_SOURCE_DIR}/embobj/plus/utils/eOtheEthLowLevelParser_hid.h
+#                                 ${CMAKE_SOURCE_DIR/can/canProtocolLib/iCubCanProto_types.h}
+  )
+
+
+  add_library(${LIBRARY_TARGET_NAME} SHARED ${${LIBRARY_TARGET_NAME}_HDR} ${${LIBRARY_TARGET_NAME}_SRC})
```

Comment:
Please do not hardcode the type of library here. If you want to build shared library by default, set (as an option) the default value of BUILD_SHARED_LIBS to ON, but do not hardcode the type of library in   the `add_library` call, see https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt#L98 . Furthermore, if you want to build as shared by default, this will fail on Windows unless you export the symbols, either by setting CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS  to ON (easier, see https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt#L45) or manually decorating all the public symbols with the export macros